webflow-hacks/misc-hacks/51/51.ts

```diff
@@ -0,0 +1,41 @@
+document.addEventListener('DOMContentLoaded', () => {
+  const TABLE_SELECTOR = '[fs-hacks-element="table"]';
+  const COLUMN_TYPE_SELECTOR = [
+    '[fs-hacks-element="design-column"]',
+    '[fs-hacks-element="development-column"]',
+    '[fs-hacks-element="subscription-column"]',
+  ];
+  const COLUMN_TOTAL_SELECTOR = [
+    '[fs-hacks-element="design-total-fee"]',
+    '[fs-hacks-element="development-total-fee"]',
+    '[fs-hacks-element="subscription-total-fee"]',
+  ];
+  // loop through all tables
+  const tables = document.querySelectorAll<HTMLDivElement>(TABLE_SELECTOR);
+  tables.forEach((table) => {
+    updateSubTotals(table, COLUMN_TOTAL_SELECTOR, COLUMN_TYPE_SELECTOR);
+  });
+});
+
+/**
+ * This function is used to update the total fees for each column of the table.
+ * @param table DOM element of the table
+ * @param displayTotalSelector selectors Column that displays the total fee
+ * @param columnValueSelectors Selectors for the inidividual fees by column
+ */
+
+function updateSubTotals(table: HTMLDivElement, displayTotalSelector: string[], columnValueSelectors: string[]) {
+  // loop through all columns.
+  for (const columnValueSelector of columnValueSelectors) {
+    let totalFee = 0;
+    const index = columnValueSelectors.indexOf(columnValueSelector);
```

Comment:
In this case it would make more sense to use `forEach` instead of `for...of` because `foreEach` provides the index in the callback.
```typescript
columnValueSelectors.forEach(( selector, index ) => {})
```

But I think you could make this code more readable by using a different structure for the selectors, something like:
```typescript
const COLUMN_SELECTORS: Array<{ type: string; total: string }> = [
  {
    type: '[fs-hacks-element="design-column"]',
    total: '[fs-hacks-element="design-total-fee"]',
  },
  {
    type: '[fs-hacks-element="development-column"]',
    total: '[fs-hacks-element="development-total-fee"]',
  },
  {
    type: '[fs-hacks-element="subscription-column"]',
    total: '[fs-hacks-element="subscription-total-fee"]',
  },
];
```

This way you don't rely on two separate Arrays that could be unordered.